src/dialects/postgres/schema/columncompiler.js

```diff
@@ -30,8 +30,19 @@ assign(ColumnCompiler_PG.prototype, {
 
   // Create the column definition for an enum type.
   // Using method "2" here: http://stackoverflow.com/a/10984951/525714
-  enu(allowed) {
-    return `text check (${this.formatter.wrap(this.args[0])} in ('${allowed.join("', '")}'))`;
+  enu(allowed, options) {
+    options = options || {};
+
+    const values = allowed.join("', '");
+
+    if (options.nativeEnum) {
+      this.pushAdditional(function() {
+        this.pushQuery(`create type ${options.enumName} as enum ('${values}')`)
```

Comment:
shouldn't enumName be wrapped to some kind of quotes?

Reply:
I guess if it is camel cased it should be, just like every other identifier. I will double check though.